src/titiler/core/titiler/core/algorithm/__init__.py

```diff
@@ -0,0 +1,59 @@
+"""titiler.core.algorithm."""
+
+from copy import copy
+from enum import Enum
+from typing import Dict, List, Type
+
+import attr
+
+from titiler.core.algorithm.base import AlgorithmMetadata, BaseAlgorithm  # noqa
+from titiler.core.algorithm.dem import Contours, HillShade, TerrainRGB, Terrarium
+from titiler.core.algorithm.index import NormalizedIndex
+
+default_algorithms: Dict[str, Type[BaseAlgorithm]] = {
+    "hillshade": HillShade,
+    "contours": Contours,
+    "normalizedIndex": NormalizedIndex,
+    "terrarium": Terrarium,
+    "terrainrgb": TerrainRGB,
+}
+
+
+@attr.s(frozen=True)
+class Algorithms:
```

Comment:
store default algorithm and allow users to register their own. The Class is then passed to the Dependency